### PR TITLE
MAT-6255: discard issue for QDM 

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-6255](https://jira.cms.gov/browse/MAT-6255)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.tsx [hunk 1]
```diff
@@ -91,7 +96,7 @@ const ElementsSection = (props: { handleTestCaseErrors: Function }) => {
 
   const [availableDataElements, setAvailableDataElements] =
     useState<DataElement[]>();
-  const [selectedDataElement, setSelectedDataElement] = useState<DataElement>();
+  // const [selectedDataElement, setSelectedDataElement] = useState<DataElement>();
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.tsx]: please remove commented code once changes are finalized

**Review by nmorasb — DISMISSED** [review]

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx [hunk 2]
```diff
@@ -359,80 +364,77 @@ describe("ElementsSection allows card opening and closing", () => {
...
+
+    expect(setSelectedDataElement).toHaveBeenCalledWith(null);
+    renderElementsSectionComponent(rerender, null);
     await waitFor(() => {
-      expect(screen.queryByTestId("data-element-card")).not.toBeInTheDocument();
+      expect(queryByTestId("data-element-card")).not.toBeInTheDocument();
     });
   });
 
   test("should open and close a data element card manual close selection", async () => {
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx]: can you rename one of these tests so they aren't both the same name

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx [hunk 3]
```diff
@@ -360,80 +364,77 @@ describe("ElementsSection allows card opening and closing", () => {
...
       "data-type-Adverse Event: Encounter Inpatient"
     );
 
-    expect(adverseEventDataType).toBeInTheDocument();
-    act(() => {
-      fireEvent.click(adverseEventDataType);
-    });
+    await userEvent.click(adverseEventDataType);
+    expect(setSelectedDataElement).toBeCalled();
+    selectedDataElement = testDataElements[11];
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx:400]: I think this `selectedDataElement` might not be used. If so please delete

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx [hunk 4]
```diff
@@ -360,80 +364,77 @@ describe("ElementsSection allows card opening and closing", () => {
...
+    const deviceDataType = getByTestId(
       "data-type-Device, Order: Cardiopulmonary Arrest"
     );
-    act(() => {
-      fireEvent.click(deviceDataType);
-    });
+
+    userEvent.click(deviceDataType);
+    expect(setSelectedDataElement).toBeCalled();
+    selectedDataElement = testDataElements[14];
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx:470]: I think this `selectedDataElement` might not be used. If so please delete

**Review by nmorasb — APPROVED** [review]

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx (hunk 4 above)
Comment (chubert-sb) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx:470]: I'll take care of this in MAT-6282
